.circleci/config.yml

```diff
@@ -38,29 +33,93 @@ jobs:
             docker-compose build frontend 
             docker-compose build py-backend
 
-  lint:
+  test:
     docker:
-      - image: cimg/python:3.9.7
+      - image: cimg/base:2021.05
     steps:
       - checkout
+      - setup_remote_docker:
+          version: 19.03.13
       - run:
-          name: Run linter checks
+          name: Create env file
           command: |
-            pip install black isort
-            echo "Running black checks..."
-            black backend/python/app/ --check
-            echo "Checking import order..."
-            isort --profile black backend/python/app/ -c
-            echo "Code properly linted!"
+            touch .env
+            echo "Writing env variables to .env"
+            env > .env
+      - run:
+          name: Running tests
+          command: |
+            set -x
+            echo "Running backend tests"
+            docker-compose -f docker-compose.testing.yml up --abort-on-container-exit
+
+  deploy:
+    docker:
+      - image: cimg/node:14.15.5
+    steps:
+      - checkout
+      - run:
+          name: Build frontend
+          command: |
+            cd frontend
+            yarn install --frozen-lockfile
+            yarn build
+      - run:
+          name: Setup Credentials
+          command: |
+            sudo apt-get update
+            sudo apt-get -qq install git-ftp
+            echo "Setting up FTP credentials..."
+            HOST=`echo ${FTP_HOST_BASE64} | base64 --decode`
+            USERNAME=`echo ${FTP_USERNAME_BASE64} | base64 --decode`
+            PASSWORD=`echo ${FTP_PASSWORD_BASE64} | base64 --decode`
+            echo "export HOST=$HOST" >> $BASH_ENV
+            echo "export USERNAME=$USERNAME" >> $BASH_ENV
+            echo "export PASSWORD=$PASSWORD" >> $BASH_ENV
+      - run:
```

Comment:
Interesting that you're deploying frontend and backend separately, any reason for this?

Reply:
I think the frontend build is supposed to be in the root of the server so the `index.html` can be served when a user goes to the website
Right now I just have the backend going to a folder called `backend/`; because they aren't being deployed in the same relative locations as our repository, I had to set them up separately